.github/workflows/pull-request-formatting-validator.yml

```diff
@@ -0,0 +1,168 @@
+# This workflow validates basic pull request formatting requirements are met.
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+name: Validate Pull Request Formatting
+
+on:
+  pull_request:
+    types:
+      - edited
+      - opened
+      - reopened
+      - synchronize
+
+# Prevent concurrent runs for the same PR
+concurrency:
+  group: pr-validation-${{ github.event.number }}
+  cancel-in-progress: true
+
+jobs:
+  validate_pr:
+    runs-on: ubuntu-latest
+
+    permissions:

```

Comment:
Are these permissions necessary if you are using the generated app token instead?

Reply:
Switched back to `pull_request_target` so the workflow no longer runs as part of this PR (since the workflow is not in `main` yet) and I updated it to exclusively use the generated token instead of the default token. `permissions` block is dropped.